cmake/modules/llext.cmake

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2024 Schneider Electric
+# SPDX-License-Identifier: Apache-2.0
+
+include_guard(GLOBAL)
+
+# Add a macro that add a llext extension.
+# macro is based on zephyr_library template
+#
+# output an elf file usable by llext
+#
+# Code can be linked by 2 differents ways:
+#     1. using partial link (-r flag in gcc and -fno-pic) (default)
+#     2. using shared link (-shared flag in gcc and -fpic)
+#
+# The code will be compiled with mostly the same C compiler flags used
+# in the Zephyr build, but with some important modifications.
+# Toolchain flag are automatically inherited and filtered.
+#
+# NOPIC:
+#     -mlong-calls flag is always added
+#     libc is detected from zephyr configuration and libc function code is embedded in extension
+# PIC:
+#    code is compiled and linked with -fpic flag
+#    libc is detected from zephyr configuration and libc function are called. They are external symbol for extension.
+#
+# Example usage:
+#     zephyr_llext(hello_world)
+#     zephyr_llext_sources(hello_world.c hello_world2.c)
+#     zephyr_llext_include_directories(.)
+# will compile the source file hello_world.c and hello_world2.c to
+# ${PROJECT_BINARY_DIR}/hello_world.llext
+#
+
+
+macro(zephyr_llext name)
```

Comment:
why is this a macro and not a function ?

All variables defined inside a macro will keep living when macro returns, which can give some nasty surprises, so extra care must be taken when deciding to define a macro.

So far I've not seen a reason why this must be implemented as a macro, so would like to understand the reasons.